### PR TITLE
[WIP] Dockers implementation

### PR DESCRIPTION
**Pull request by someanon**
Don't know how to properly test this. @AndrewSamokhvalov can you help me?

### CODE REVIEW

File: bitcoind/btcjson/help_test.go [hunk 1]
```diff
@@ -565,7 +565,7 @@ func TestHelpArgInternals(t *testing.T) {
 
 // TestMethodHelp ensures the method help function works as expected for various
 // command structs.
-func TestMethodHelp(t *testing.T) {
```
Comment (AndrewSamokhvalov) [inline bitcoind/btcjson/help_test.go]: Why we need this?

File: metrics/crypto/backend_test.go [hunk 2]
```diff
@@ -9,14 +9,16 @@ type testBackend struct {
 	panicSent bool
 }
 
-func (b *testBackend) AddRequest(daemon, asset, request string)         {}
```
Comment (AndrewSamokhvalov) [inline metrics/crypto/backend_test.go]: Why we need this?

File: bitcoind/btcjson/help_test.go (hunk 1 above)
Comment (someanon) [reply bitcoind/btcjson/help_test.go]: Was error about "already have function with this name"

File: metrics/crypto/backend_test.go (hunk 2 above)
Comment (someanon) [reply metrics/crypto/backend_test.go]: Was error about "not implementing"

File: common/connector.go [hunk 3]
```diff
@@ -124,3 +128,92 @@ type LightningConnector interface {
...
 }
+
+type PaymentsStore interface {
+	Payment(id string) (*Payment, error)
+	AddPayment(p *Payment) error
+}
+
+var PaymentNotFound = errors.New("payment not found")
+
+type MemoryPaymentsStore struct {
```
[the remark this answers is not in the document]
Comment (someanon) [reply common/connector.go]: Yes, it should. I'm using `StartCleaner` and `StopCleaner` in main.

File: crpc/go/server.go [hunk 4]
```diff
@@ -430,3 +433,25 @@ func (s *Server) Estimate(_ context.Context,
...
+// PaymentReceived is used to determine if payment with given ID is received
+func (s *Server) PaymentReceived(_ context.Context,
+	req *PaymentReceivedRequest) (*PaymentReceivedResponse, error) {
+
+	log.Tracef("command(%v), request(%v)", getFunctionName(), spew.Sdump(req))
+
+	_, err := s.paymentsStore.Payment(req.Id)
+
+	if err != nil && err != common.PaymentNotFound {
+		return nil, newErrInternal(err.Error())
```
[the remark this answers is not in the document]
Comment (someanon) [reply crpc/go/server.go]: Yes, should be public error

File: crpc/generate.sh [hunk 5]
```diff
@@ -5,39 +5,3 @@ protoc -I/usr/local/include -I. \
        -I$GOPATH/src \
```
[the remark this answers is not in the document]
Comment (someanon) [reply crpc/generate.sh]: spltted

File: main.go [hunk 6]
```diff
@@ -304,7 +309,10 @@ func backendMain() error {
 						return
 					case payments := <-client.ReceivedPayments():
 						for _, payment := range payments {
-							doDeposit(engine, payment, asset)
+							if isExchangePayment(payment) {
```
[the remark this answers is not in the document]
Comment (someanon) [reply main.go]: added

**Comment by someanon** [comment]
We need to test it along with new zigzag

**Comment by someanon** [comment]
Reworked PR, removed all changes related to exhcnage and zigzag shared usage.

**Pull request closed without merging**